### DISCUSSION TITLE
Compatibility with MDsveX

### DISCUSSION THREAD

**Discussion by mateoroldos** [sveltekit-i18n/lib#67] [Q&A]
Hi everyone!

Does anyone have any experience dealing with i18n library + using markdown with a library like MDsveX.
Would be very helpful if someone can share his experience + any example of this libraries combined.

Thanks!

**Comment by jarda-svoboda** [comment]
Hi @mateoroldos!

I think that currently there is no way how to easily achieve what you need. Anyway, i'm planning to [introduce connectors](https://github.com/sveltekit-i18n/lib/issues/64) in the next major version which should work for direct text translations. I guess this is exactly your use case?

Currently the whole idea of direct translations is in "forming" phase, so feel free to comment related issue with ideas you have or description how would you like to use it. Basically, it should take some markup (html, free text, md?, ... ) and automatically translate it using some service such as Google Translation...

**Reply by jarda-svoboda** [reply]
On the other hand if you are about using `sveltekit-i18n` lib inside `svx` files, it should work normally, I guess…

**Reply by saschakrueger** [reply]
Would it be possible to see an example of this?

**Comment by jarda-svoboda** [comment]
Closing the loop on this thread, since my answer above is out of date on both halves.

**Connectors will not ship.** The idea was tracked in #64 and was closed as not planned during the v3 triage (#214). The reasoning, so the decision is not just a silent close:

- Runtime machine translation would put a call to a third-party service on the render path. This library has zero runtime dependencies and performs no network access of its own — fetching is the consumer's loader, deliberately. A connector inverts that.
- There is nothing to build. A loader is `async () => data`, so it may already call a translation API, a CMS, or anything else. A connector would add public surface for what the existing surface does.
- Machine-translated UI strings without review break exactly what a message format exists to handle: plurals, gendered forms, placeholder order, string length.
- If machine translation belongs anywhere, it is build time — pre-translate the files, review the diff, commit the result. That is separate tooling, not runtime behaviour.

On the original MDsveX question, the two halves have different answers:

- **Translating the markdown content itself** is content localisation, not message translation: per-locale files (`en/about.svx`, `cs/about.svx`) picked by the route. This library translates messages and is not meant to be a document pipeline, so there is no feature here to wait for.
- **Using the library inside `.svx` files** should work, since mdsvex compiles `.svx` into a Svelte component and `t` is then used as in any other component. That path is not covered by our examples, so I would not treat it as guaranteed until one exists — v3 in particular replaces the store surface with runes, so the `$t` form in v2 becomes `t(...)`. Reworking the examples is tracked in #230; I will make sure an `.svx` case is considered there.
